src/server/game/Entities/Player/Player.cpp

```diff
@@ -11634,12 +11634,15 @@ Item* Player::_StoreItem(uint16 pos, Item* pItem, uint32 count, bool clone, bool
     }
 }
 
-Item* Player::EquipNewItem(uint16 pos, uint32 item, bool update)
+Item* Player::EquipNewItem(uint16 pos, uint32 item, bool update, int32 randomPropertyId)
 {
     if (Item* pItem = Item::CreateItem(item, 1, this))
     {
         ItemAddedQuestCheck(item, 1);
         UpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_RECEIVE_EPIC_ITEM, item, 1);
+        UpdateAchievementCriteria(ACHIEVEMENT_CRITERIA_TYPE_OWN_ITEM, item, 1);
```

Comment:
and what's that ninja addition here?

Reply:
Fixes a very edge case where items bought from a vendor directly into an equip slot (without going to bag first) didn't count towards ACHIEVEMENT_CRITERIA_TYPE_OWN_ITEM, which it should be